src/core/scene.ts

```diff
@@ -1363,4 +1370,281 @@ export class Scene {
+   *
+   * ```javascript
+   *
+   * myScene.removePreset(lastPreset)
+   * .then(isRemoved => {
+   *   console.log('preset is removed : ' + isRemoved)
+   * });
+   * ```
+   */
+  removePreset(preset: string): Promise<boolean> {
+    return new Promise((resolve, reject) => {
+      if (Environment.isSourcePlugin()) {
+        reject(Error('Not supported on source plugins'));
+      } else if (versionCompare(getVersion()).is.lessThan(scenePresetsVersion)) {
+        reject(Error('Not supported in this XBC version'));
+      } else if (preset === '{00000000-0000-0000-0000-000000000000}') {
+        reject(Error('Cannot delete the default preset'));
+      } else {
+        iApp.set('sceneremovepreset:' + this._uid, preset).then(value => {
+          resolve(value);
+        });
+      }
+    })
+  }
+
+  /**
+   * return: Promise<string>
+   *
+   * Get the preset transition easing function for the scene
+   *
+   * #### Usage
+   *
+   * ```javascript
+   * myScene.getPresetTransition().then(function(presetTransition) {
+   *  console.log('Preset transition is ' + presetTransition);
+   * });
+   * ```
+   */
+  getPresetTransitionEasing(): Promise<string> {
+    return new Promise((resolve, reject) => {
+      if (Environment.isSourcePlugin()) {
+        reject(Error('Not supported on source plugins'));
+      } else if (versionCompare(getVersion()).is.lessThan(scenePresetsVersion)) {
+        reject(Error('Not supported in this XBC version'));
+      } else {
+        iApp.get('scenepresettransitionfunc:' + this._uid).then(value => {
+          if (value === '') {
+            value = 'none';
+          }
+          resolve(value);
+        });
+      }
+    });
+  }
+
+  /**
+   * param: (presetTransitionEasing: string)
+   * ```
+   * return: Promise<boolean>
+   * ```
+   * Switch to the specified preset transition easing function for the scene
+   * Possible values ('' or 'none', 'easeInCubic', 'easeOutCubic', 'easeInOutCubic')
+   *
+   * #### Usage
+   *
+   * ```javascript
+   *
+   * myScene.setPresetTransitionEasing().then(function(presetTransition) {
+   *  console.log('Preset transition is ' + presetTransition);
+   * });
+   * ```
+   */
+  setPresetTransitionEasing(presetTransitionEasing: string): Promise<boolean> {
+    return new Promise((resolve, reject) => {
+      if (Environment.isSourcePlugin()) {
+        reject(Error('Not supported on source plugins'));
+      } else if (versionCompare(getVersion()).is.lessThan(scenePresetsVersion)) {
+        reject(Error('Not supported in this XBC version'));
+      } else if (supportedPresetTransitionEasingFunctions.indexOf(presetTransitionEasing) < 0) {
+        reject(Error('Easing function not supported for preset transitions'));
+      } else {
+        presetTransitionEasing = presetTransitionEasing === 'none' ? '' : presetTransitionEasing;
+        iApp.set('scenepresettransitionfunc:' + this._uid, presetTransitionEasing).then(value => {
+          resolve(value);
```

Comment:
Do we need to have a `if (value === '')` here just like the one in `getPresetTransitionEasing`?

Reply:
Basically, `'none'` here is an alias for `''`. The correct internal value is `''`, which is quite weird so we use `'none'` externally instead. As for the resolved value, that is simply `true` or `false`, depending if the setting is successful or not.

Reply:
I see, we might want to update the example code in the comment above then. It gave the impression that it would resolve with a string.

Reply:
Shoot, it was the exact example as getPresetTransitionEasing. My bad.